### DISCUSSION TITLE
glib

### DISCUSSION THREAD

**Discussion by fswarbrick** [zopencommunity/meta#824] [Porting]
Can applications using glib (glibc?) be ported?  If so, how?

**Comment by IgorTodorovskiIBM** [comment]
Glib and Glibc are two different libraries. 

- glib is a utility library for C - https://docs.gtk.org/glib/. We have not ported this as part of z/OS Open Tools, but we could if there's a need for it.

- glibc is the standard C runtime library for Linux. LE's C runtime library provides most of this support, but we also use zoslib, gnulib and/or libbsd to fill in the gaps. Technically this could be ported, as was done by ambitus here: https://github.com/ambitus/glibc. That port required Metal C and ELF format objects, though.

**Reply by fswarbrick** [reply]
Here's the error I am getting:
```
checking for glib-2.0 >= 2.32... no
configure: error: glib-2.0 not found or version too old (must be >= 2.32)
```
The docs say both "glibc" and "glib2 >= 2.32 are required, but based on what you say above I'm thinking the LE probably (hopefully) fulfills the glibc requirement.
The application, by the way, is Midnight Commander.
